### PR TITLE
Feat/tokengate v2

### PR DESCRIPTION
**Pull request by valentinludu**

### CODE REVIEW

File: components/common/TokenGateModal/utils/getTokensUnifiedAccessControlConditions.ts [hunk 1]
```diff
@@ -1,46 +1,38 @@
...
-        }
+        type: 'ERC20',
+        tokenIds: [],
+        quantity: amount
       }
     ];
   }
 
-  if (check === 'token') {
+  if (check === 'token' && contract) {
```
Comment (Devorein) [inline components/common/TokenGateModal/utils/getTokensUnifiedAccessControlConditions.ts:26]: contract is not used in `contractAddress` section. Not sure we should be adding it to the condition

File: lib/blockchain/readContract.ts
Comment (Devorein) [inline lib/blockchain/readContract.ts:1]: This doesn't seem to be used anywhere

**Review by Devorein — APPROVED** [review]

**Review by Devorein — APPROVED** [review]

File: components/common/TokenGateModal/utils/getTokensUnifiedAccessControlConditions.ts (hunk 1 above)
Comment (valentinludu) [reply components/common/TokenGateModal/utils/getTokensUnifiedAccessControlConditions.ts:26]: Good point. No we don't need it.

File: lib/blockchain/readContract.ts
Comment (valentinludu) [reply lib/blockchain/readContract.ts:1]: Deleted!

**Pull request merged**